snuba/replacer.py

```diff
@@ -20,15 +19,14 @@
 
 class ReplacerWorker(AbstractBatchWorker[KafkaPayload, Replacement]):
     def __init__(
-        self, clickhouse: ClickhousePool, dataset: Dataset, metrics: MetricsBackend
+        self, clickhouse: ClickhousePool, storage: WritableTableStorage, metrics: MetricsBackend
     ) -> None:
         self.clickhouse = clickhouse
         self.metrics = metrics
-        processor = enforce_table_writer(dataset).get_replacer_processor()
-        dataset_name = get_dataset_name(dataset)
+        processor = storage.get_table_writer().get_replacer_processor()
         assert (
             processor
-        ), f"This dataset writer does not support replacements {dataset_name}"
+        ), f"This storage writer does not support replacements {storage}"
```

Comment:
Since this is not the storage name could you please use type(storage) in the message.

Reply:
Yeah. Will likely need a way to get the name in future.